File: 백준/Silver/5800. 성적 통계/성적 통계.cc
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() 
{
    int num;
    cin >> num;

    for (int i = 0; i < num; i++) {
        int student_count;
        cin >> student_count;
        int gap = -1;

        vector<int> v;
        for (int j = 0; j < student_count; j++) {
            int score;
            cin >> score;
            v.push_back(score);
        }

        // 내림차순 정렬
        sort(v.begin(), v.end(), greater<>());
        int max = v.front();
        int min = v.back();

        for (int j = 0; j < student_count - 1; j++) {
            if (gap < v[j] - v[j + 1]) {
                gap = v[j] - v[j + 1];
            }
        }
        cout << "Class " << i + 1 << endl;
        cout << "Max " << max << ", Min " << min << ", Largest gap " << gap << endl;

    }
    return 0;
}

File: 백준/Silver/5800. 성적 통계/성적 통계.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        int K = sc.nextInt();

        for (int i = 0; i < K; i++) {
            System.out.println("Class " + (i + 1));

            int N = sc.nextInt();
            int[] arr = new int[N];
            for (int j = 0; j < N; j++) {
                arr[j] = sc.nextInt();
            }
            
            Arrays.sort(arr);

            int largestGap = 0;
            for (int j = 1; j < N; j++) {
                largestGap = Math.max(largestGap, arr[j] - arr[j - 1]);
            }
            System.out.println("Max " + arr[N - 1] + ", Min " + arr[0] + ", Largest gap " + largestGap);
        }
    }
}